### PR TITLE
ENH API tests and fixes

### PR DESCRIPTION
**Pull request by beckermr**
This PR has a bunch of API tests and fixes. I noticed a lot of these things when working on interpolated images and decided to move this to a new PR.

### CODE REVIEW

File: jax_galsim/wcs.py [hunk 1]
```diff
@@ -849,11 +849,12 @@ def _profileToImage(self, world_profile, flux_ratio, offset):
     def _pixelArea(self):
         return abs(self._det)
 
+    # @_wraps(_galsim.JacobianWCS.getMatrix, lax_description="The JAX version returns JAX arrays.")
```
Comment (beckermr) [inline jax_galsim/wcs.py:852]:
```suggestion
```

**Comment by beckermr** [comment]
Can we merge this one @ismael-mendoza?

**Comment by ismael-mendoza** [comment]
I'll post my review shortly @beckermr

File: tests/jax/test_temporary_tests.py
Comment (ismael-mendoza) [inline tests/jax/test_temporary_tests.py:1]: it seems that you run `isort` in a subset of the test files, did you want to standarize them all and add this check to the CI ?

**Review by ismael-mendoza — COMMENTED** [review]
Just a few minor comments and questions, but should be almost ready to merge

File: tests/jax/test_temporary_tests.py
Comment (ismael-mendoza) [reply tests/jax/test_temporary_tests.py:1]: (doesnt' have to be in this PR)

File: jax_galsim/transform.py [hunk 2]
```diff
@@ -321,7 +325,7 @@ def _kValue(self, kpos):
     def _drawReal(self, image, jac=None, offset=(0.0, 0.0), flux_scaling=1.0):
         dx, dy = offset
         if jac is not None:
-            x1 = jac.dot(self.offset.array)
+            x1 = jac.dot(self.offset._array)
```
[the remark this answers is not in the document]
Comment (ismael-mendoza) [reply jax_galsim/transform.py:328]: ah my bad, confused it with the input argument

File: jax_galsim/moffat.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ def body(i, xcur):
         x = jnp.sqrt(x - 1)
         return re / x
 
-    rd = jax.lax.fori_loop(0, Nloop, body, re)
+    rd = jax.lax.fori_loop(0, 1000, body, re)
```
[the remark this answers is not in the document]
Comment (beckermr) [reply jax_galsim/moffat.py:65]: This default is never changed in the code so I assume it is always ok. Idk why it is the default.

File: jax_galsim/wcs.py [hunk 4]
```diff
@@ -850,10 +850,10 @@ def _pixelArea(self):
         return abs(self._det)
 
     def getMatrix(self):
-        """Get the Jacobian as a JAX matrix:
+        """Get the Jacobian as a NumPy matrix:
```
[the remark this answers is not in the document]
Comment (beckermr) [reply jax_galsim/wcs.py:853]: This is to keep the same doc string. We can wrap it, but the code as it stands now doesn't consistently use one or the other.

File: jax_galsim/moffat.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ismael-mendoza) [reply jax_galsim/moffat.py:65]: ah ok, I'll ask @jecampagne but probably ok to keep this for this PR

File: tests/jax/test_temporary_tests.py
Comment (beckermr) [reply tests/jax/test_temporary_tests.py:1]: Idk how this happened actually. In any case it was not intentional so I think we can just leave it and move on.

File: jax_galsim/wcs.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ismael-mendoza) [reply jax_galsim/wcs.py:853]: ok! probably leave for later to apply wrapping consistently if possible

**Review by ismael-mendoza — APPROVED** [review]

File: jax_galsim/position.py [hunk 5]
```diff
@@ -127,7 +140,7 @@ def shear(self, shear):
             a `galsim.PositionD` instance.
         """
         shear_mat = shear.getMatrix()
-        shear_pos = jnp.dot(shear_mat, self.array)
+        shear_pos = jnp.dot(shear_mat, jnp.array([self.x, self.y]))
```
Comment (beckermr) [inline jax_galsim/position.py:143]:
```suggestion
        shear_pos = jnp.dot(shear_mat, self._array)
```

**Pull request merged**